code/ui/manager/src/components/layout/Layout.tsx

```diff
@@ -0,0 +1,144 @@
+import { useImmerReducer } from 'use-immer';
+import type { Dispatch, MutableRefObject } from 'react';
+import React, { useEffect, useRef } from 'react';
+import { DESKTOP, getGridTemplate, MOBILE, SHARED } from './Layout.styles';
+import type { Props, LayoutState, ExposedLayoutState } from './Layout.types';
+import { MobileControls } from './Layout.MobileControls';
+import { DesktopControls } from './Layout.DesktopControls';
+
+export const Layout = ({ state: incomingState, persistence, setState, ...slots }: Props) => {
+  const [state, updateState] = useImmerReducer<LayoutState, Partial<LayoutState>>(
```

Comment:
It feels like a little bit of a bummer to add a nearly-one-megabyte dependency just for this one use.  Can you talk a little about why you chose to use it here?  Aside from the added filesize, it's also one extra tool that contributors may need to learn.  I am not familiar with it myself, and while I'm happy to learn it if it adds a ton of value, if there's a way we could avoid it, I'd prefer to, personally.

Reply:
Agree with @IanVS on this one! 👍

Reply:
I love immer, and think we should use it everywhere, but I was able to downgrade the vanilla useReducer.
Where did you get the 1MB number from?